met-web/src/components/common/Modals/ConfirmModal.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { Grid, Stack, useMediaQuery, Theme } from '@mui/material';
+import { modalStyle, PrimaryButton, SecondaryButton, MetHeader1, MetBody } from 'components/common';
+import { ModalProps } from './types';
+
+const ConfirmModal = ({ header, subText, handleClose, buttons }: ModalProps) => {
+    const isSmallScreen = useMediaQuery((theme: Theme) => theme.breakpoints.down('sm'));
+    return (
+        <Grid
+            container
+            direction="row"
+            justifyContent="flex-start"
+            alignItems="space-between"
+            sx={{ ...modalStyle }}
+            rowSpacing={2}
+        >
+            <Grid container item xs={12}>
+                <Grid item xs={12}>
+                    <MetHeader1 sx={{ fontWeight: 'bold', mb: 2 }}>{header}</MetHeader1>
+                </Grid>
+            </Grid>
+            <Grid container direction="row" item xs={12}>
+                {subText.map((text: string) => (
+                    <Grid item xs={12}>
+                        <MetBody sx={{ mb: 1 }}> {text} </MetBody>
+                    </Grid>
+                ))}
+                <Grid
+                    item
+                    container
+                    direction={{ xs: 'column', sm: 'row' }}
+                    xs={12}
+                    justifyContent="flex-end"
+                    spacing={1}
+                    sx={{ mt: '1em' }}
+                >
+                    <Stack direction={{ xs: 'column', sm: 'row' }} spacing={1} width="100%" justifyContent="flex-end">
+                        {isSmallScreen ? (
+                            <>
+                                <PrimaryButton
+                                    onClick={() => buttons[0].buttonFunction}
```

Comment:
I think instead of sending an array and indexing it like this, which is a bit ambiguous. Let's do something similar to MetWidget in the common components. Notice there the handle functions are props, if they are passed we render the corresponding button and if not we hide. Let button text be standard and static. Let that not be a variable since it would make things more difficult.

Reply:
Addressed all comments except changing the data type since not every modal has the same data (the handle confirm is not present in both modals)  thank you for all the feedback